plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -404,8 +412,12 @@ func (monitor *DirectoryMonitor) Init() error {
 		}
 	}
 
-	monitor.filesDropped = selfstat.Register("directory_monitor", "files_dropped", map[string]string{})
-	monitor.filesProcessed = selfstat.Register("directory_monitor", "files_processed", map[string]string{})
+	tags := map[string]string{
+		"directory": monitor.Directory,
+	}
+	monitor.filesDropped = selfstat.Register("directory_monitor", "files_dropped", tags)
+	monitor.filesProcessed = selfstat.Register("directory_monitor", "files_processed", tags)
+	monitor.filesQueued = selfstat.Register("directory_monitor", "files_queue", tags)
```

Comment:
Instead of the "hack" above I would really rather see something like this:
```suggestion
	tags := map[string]string{
		"directory": monitor.Directory,
	}
	monitor.filesDropped = selfstat.Register("directory_monitor", "files_dropped", map[string]string{})
	monitor.filesProcessed = selfstat.Register("directory_monitor", "files_processed", map[string]string{})
	monitor.filesDroppedDir = selfstat.Register("directory_monitor", "files_dropped_per_dir", tags)
	monitor.filesProcessedDir = selfstat.Register("directory_monitor", "files_processed_per_dir", tags)
	monitor.filesQueuedDir = selfstat.Register("directory_monitor", "files_queue_per_dir", tags)
```

Reply:
So the resulting metrics would become this?

- internal_directory_monitor
  - fields:
    - files_processed - How many files have been processed (counter)
    - files_dropped - How many files have been dropped (counter)
- internal_directory_monitor
  - tags:
    - directory - The monitored directory
  - fields:
    - files_processed_per_dir - How many files have been processed (counter)
    - files_dropped_per_dir - How many files have been dropped (counter)
    - files_queue - How many files to be processed (gauge)

I don't like the per dir field suffix naming..

Reply:
@srebhan?
> I don't like the per dir field suffix naming of the second metric..